File: where-for-dinner/where-for-dinner-crawler/src/main/java/com/java/example/tanzu/wherefordinner/model/Availability.java
package com.java.example.tanzu.wherefordinner.model;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
public class Availability 
{	
	private String diningName;

	private String address;
	
	private String locality;
	
	private String region;
	
	private String postalCode;
	
	private String phoneNumber;
	
	private String reservationURL;
	
	private List<AvailabilityWindow> availabilityWindows = new ArrayList<>();
	
	@Data
	@AllArgsConstructor
	public static class AvailabilityWindow
	{
		private long startTime;
		
		private long endTime;
	}
}

File: where-for-dinner-initialized/where-for-dinner-ui/src/Availability.js
import TimeWindow from "./TimeWindow";

function Availability(props)
{
    const timeWindows = props.avail.availabilityWindows.map((win, index)=>{
        return <TimeWindow id={index} key={index} timeWin={win}/>
    });

    return(
        <div className="availability">
            <label className="diningLabel">{props.avail.diningName}</label><br/>
            <label className="generalLabel">Address: </label>
            <label>{props.avail.address + '  ' + props.avail.locality + ',' + props.avail.region + ' ' + props.avail.postalCode}</label>
            <br/>
            <label className="generalLabel"> Phone Number:</label>
            <label>{props.avail.phoneNumber}</label><br/>
            <a className="generalLabel" href={props.avail.reservationURL} target="_blank" rel="noopener noreferrer">Reservations</a>
            {timeWindows.length ? <div><label className="timeWindowHeader">Availability</label>{timeWindows}</div> : <div><label className="noTimeWindows">NO AVAILABILITY</label></div>}
            <p/>
        </div>
    )
}

export default Availability;